third_party/amd/lib/TritonAMDGPUToLLVM/Utility.h

```diff
@@ -10,9 +10,13 @@
 #include "triton/Conversion/TritonGPUToLLVM/Utility.h"
 namespace mlir::LLVM::AMD {
 
-const char Predicated_Load[] = "__predicated_load";
-const char Predicated_Load_NT[] = "__predicated_load_NT";
-const char Predicated_Store[] = "__predicated_store";
+const std::string Predicated_Load = "__predicated_load";
```

Comment:
Normal `const char` works here? Prefer to avoid non-POD global variables.

Also while we are here--can you change the variables to follow naming convention?

Reply:
The purpose of using string type here is willing to have specific load/store types being build base on the original ``__predicated_load/__predicated_store`` string. Const char can't be concatinated with no crutches like.

```
#define STR1 "str1"
#define STR2 "str2"

const char str1[] = STR1;
const char str2[] = STR1;
const char strConcat[] = STR1STR2;
```
String looks better IMHO.

Reply:
I'm not so sure--is the purpose to just avoid the duplication? I don't think repeating that few characters is a big issue here.

The runtime cost is very clear though: strings are non-POD and they have construction/destruction implications which happens at program startup/shutdown time due to the lifetime. And because we are building one string on top of others, it additionally has implications of initialization orders. Generally non-POD in globals a bit tricky esp. related to shared libaries and such. We are not doing that here but just wanted to point out the trickiness like [this](https://stackoverflow.com/questions/1538137/c-static-global-non-pod-theory-and-practice) and [this](https://stackoverflow.com/questions/1538137/c-static-global-non-pod-theory-and-practice).

Also every time we are trying to use these strings we are likely have silent copy the string rather than just copy a simple `char*`.

Due to the above reasons, I think the cost of trying to avoid duplication using strings far outweigh the benefits there.